apps/meteor/app/apps/server/bridges/messages.ts

```diff
@@ -44,30 +46,33 @@ export class AppMessageBridge extends MessageBridge {
 			throw new Error('A message must exist to update.');
 		}
 
-		const msg = await this.orch.getConverters()?.get('messages').convertAppMessage(message);
+		// #TODO: #AppsEngineTypes - Remove explicit types and typecasts once the apps-engine definition/implementation mismatch is fixed.
+		const msg: IMessage | undefined = await this.orch.getConverters()?.get('messages').convertAppMessage(message);
 		const editor = await Users.findOneById(message.editor.id);
 
 		if (!editor) {
 			throw new Error('Invalid editor assigned to the message for the update.');
 		}
 
-		await updateMessage(msg, editor);
+		await updateMessage(msg as IMessage, editor);
 	}
 
-	protected async delete(message: IMessage, user: IUser, appId: string): Promise<void> {
+	protected async delete(message: IAppsMessage, user: IAppsUser, appId: string): Promise<void> {
 		this.orch.debugLog(`The App ${appId} is deleting a message.`);
 
 		if (!message.id) {
 			throw new Error('Invalid message id');
 		}
 
-		const convertedMsg = await this.orch.getConverters()?.get('messages').convertAppMessage(message);
-		const convertedUser = await this.orch.getConverters()?.get('users').convertById(user.id);
+		// #TODO: #AppsEngineTypes - Remove explicit types and typecasts once the apps-engine definition/implementation mismatch is fixed.
+		const convertedMsg: IMessage | undefined = await this.orch.getConverters()?.get('messages').convertAppMessage(message);
+		const convertedUser: IAppsUser | undefined = await this.orch.getConverters()?.get('users').convertById(user.id);
 
-		await deleteMessage(convertedMsg, convertedUser);
+		// #TODO: #AppsEngineTypes - deleteMessage expects an IUser but is receiving an IAppsUser - this is probably a bug
+		await deleteMessage(convertedMsg as IMessage, convertedUser as unknown as IUser);
```

Comment:
This one is probably a bug; The deleteMessage function expects an IUser but is receiving an IAppsUser.

Reply:
Yes, looks like a potential bug. It might work now due to both types sharing some properties, but that's just luck.